src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/CompilerToVM.java

```diff
@@ -1148,14 +1159,23 @@ ResolvedJavaMethod[] getDeclaredConstructors(HotSpotResolvedObjectTypeImpl klass
     native ResolvedJavaMethod[] getDeclaredConstructors(HotSpotResolvedObjectTypeImpl klass, long klassPointer);
 
     /**
-     * Gets the {@link ResolvedJavaMethod}s for all the non-constructor methods of {@code klass}.
+     * Gets the {@link ResolvedJavaMethod}s for all non-overpass instance methods of {@code klass}.
      */
     ResolvedJavaMethod[] getDeclaredMethods(HotSpotResolvedObjectTypeImpl klass) {
         return getDeclaredMethods(klass, klass.getKlassPointer());
     }
 
     native ResolvedJavaMethod[] getDeclaredMethods(HotSpotResolvedObjectTypeImpl klass, long klassPointer);
 
+    /**
+     * Gets the {@link ResolvedJavaMethod}s for all instance methods of {@code klass}.
```

Comment:
instance -> non-static
Instance -> NonStatic

Reply:
thinking about it more, it's probably better if we do no filter and return all methods in `InstanceKlass->_methods`. How about:

```   
 /**
     * Returns a list containing all methods present within this type. This list can include
     * methods implicitly created and used by the VM.
     * The returned List is unmodifiable; calls to any mutator method
     * will always cause {@code UnsupportedOperationException} to be thrown.
     *
     * @param forceLink if {@code true}, forces this type to be {@link #link linked}
     */
    List<ResolvedJavaMethod> getAllMethods(boolean forceLink);
```

Reply:
Yes, that's a good idea - it's more future proof and lets the caller do the filtering.

`This list can include methods implicitly created and used by the VM that are not present in {@link #getDeclaredMethods}.`